### PR TITLE
Terraform Resources for Router Firewall Groups

### PR DESCRIPTION
**Pull request by manjunath-batakurki**

### CODE REVIEW

**Comment by manjunath-batakurki** [comment]
Test Logs :

```
manjuna@ubuntu:~/hpe-hcss/terraform$ TF_REATTACH_PROVIDERS='{"terraform.example.com/vmaas/hpegl":{"Protocol":"grpc","ProtocolVersion":5,"Pid":55017,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin3063071585"}}}' terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group will be created
  + resource "hpegl_vmaas_router_firewall_rule_group" "tf_router_firewall_rule_group" {
      + description   = "Router Firewall rule group created via terraform"
      + external_type = "GatewayPolicy"
      + group_layer   = "LocalGatewayRules"
      + id            = (known after apply)
      + is_deprecated = (known after apply)
      + name          = "tf_router_firewall_group"
      + priority      = 120
      + router_id     = 39
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group: Creating...
hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group: Creation complete after 3s [id=31]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
manjuna@ubuntu:~/hpe-hcss/terraform$ TF_REATTACH_PROVIDERS='{"terraform.example.com/vmaas/hpegl":{"Protocol":"grpc","ProtocolVersion":5,"Pid":55017,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin3063071585"}}}' terraform destroy
hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group: Refreshing state... [id=31]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group will be destroyed
  - resource "hpegl_vmaas_router_firewall_rule_group" "tf_router_firewall_rule_group" {
      - description   = "Router Firewall rule group created via terraform" -> null
      - external_type = "GatewayPolicy" -> null
      - group_layer   = "LocalGatewayRules" -> null
      - id            = "31" -> null
      - name          = "tf_router_firewall_group" -> null
      - priority      = 120 -> null
      - router_id     = 39 -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group: Destroying... [id=31]
hpegl_vmaas_router_firewall_rule_group.tf_router_firewall_rule_group: Destruction complete after 1s

Destroy complete! Resources: 1 destroyed.
manjuna@ubuntu:~/hpe-hcss/terraform$
```

File: internal/cmp/router_firewall_rule_group.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	// if router not found set is_deprecated flag=true
+	if router.ID == 0 {
+		log.Printf("[ERROR] Router with %d id is not found on Firewall Rule Group plan", tfFirewallRuleGroup.RouterID)
+		tfFirewallRuleGroup.IsDeprecated = true
+
+		return tftags.Set(d, tfFirewallRuleGroup)
+	}
+
+	_, err = r.routerFirewallRuleGroupClient.GetSpecificRouterFirewallRuleGroup(ctx, tfFirewallRuleGroup.RouterID,
+		tfFirewallRuleGroup.ID)
```
Comment (tshihad) [inline internal/cmp/router_firewall_rule_group.go:45]: not capturing response is fine only if we don't have any computed field from the response (other than `id`). I think here this approch is fine

File: internal/cmp/router_nat.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func (r *routerNat) Read(ctx context.Context, d *utils.Data, meta interface{}) e
 		return tftags.Set(d, tfNat)
 	}
 
-	_, err = r.routerNatClient.GetSpecificRouterNat(ctx, tfNat.ID, tfNat.ID)
+	_, err = r.routerNatClient.GetSpecificRouterNat(ctx, tfNat.RouterID, tfNat.ID)
```
Comment (tshihad) [inline internal/cmp/router_nat.go:44]: Good catch

File: internal/resources/resource_router_firewall_rule_groups.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Description:      "Firewall rule group priority",
+				ValidateDiagFunc: validations.IntAtLeast(1),
+			},
+			"external_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				InputDefault: "GatewayPolicy",
+				Description:  "Platform / vendor specific type. Pass `GatewayPolicy`.",
```
Comment (tshihad) [inline internal/resources/resource_router_firewall_rule_groups.go:43]: unwanted spance on either side of `'/'`. Should it be `Platform/vendor`??

**Review by tshihad — COMMENTED** [review]
Minor comments, please check

**Pull request closed without merging**